src/db.rs

```diff
@@ -56,6 +57,47 @@ const KEEP_LOGS: usize = 16;
 /// Value is just a vector of bytes. Value sizes up to 4Gb are allowed.
 pub type Value = Vec<u8>;
 
+#[derive(Debug, Clone, PartialEq, Eq, PartialOrd, Ord)]
+pub struct ValuePtr(Arc<Value>);
```

Comment:
Since this is used for both the key and the value, it should be named in a more generic way.  Something like `SharedVec` or `ArcVec` or `ArcBuf`